engine/execution/ingestion/core.go

```diff
@@ -120,42 +169,30 @@ func (e *Core) OnCollection(col *flow.Collection) {
 	}
 }
 
-func (e *Core) launchWorkerToConsumeThrottledBlocks() {
-	// processables are throttled blocks
-	processables := make(chan flow.Identifier, 10000)
-
+func (e *Core) launchWorkerToConsumeThrottledBlocks(ctx irrecoverable.SignalerContext) {
 	// running worker in the background to consume
 	// processables blocks which are throttled,
 	// and forward them to the block queue for processing
-	e.unit.Launch(func() {
-		e.log.Info().Msgf("starting worker to consume throttled blocks")
-		err := e.forwardProcessableToHandler(processables)
-		if err != nil {
-			e.log.Fatal().Err(err).Msg("fail to process block")
-		}
-	})
-
-	e.log.Info().Msg("initializing throttle engine")
-
-	err := e.throttle.Init(processables)
-	if err != nil {
-		e.log.Fatal().Err(err).Msg("fail to initialize throttle engine")
-	}
-
-	e.log.Info().Msgf("throttle engine initialized")
-}
-
-func (e *Core) forwardProcessableToHandler(
-	processables <-chan flow.Identifier,
-) error {
-	for blockID := range processables {
-		err := e.onProcessableBlock(blockID)
-		if err != nil {
-			return fmt.Errorf("could not process block: %w", err)
+	e.log.Info().Msgf("starting worker to consume throttled blocks")
+	defer func() {
+		e.log.Info().Msgf("worker to consume throttled blocks stopped")
+	}()
+	for {
+		select {
+		case <-ctx.Done():
+			e.log.Info().Msgf("irrecoverable.Done() is called")
+			return
+
+		case blockID := <-e.processables:
```

Comment:
Good point. 

I added a Done method to e.throttle to stop sending more blocks to the channel. And After that, let `Core` close and drain the `e.processables` channel, since Core is the creator of the `e.processables` channel.